step-functions/step-functions-handler/src/main/java/step/functions/handler/AbstractFunctionHandler.java

```diff
@@ -141,84 +137,97 @@ protected <T> T runInContext(String branch, Callable<T> callable) throws Excepti
 	 * 
 	 * @param classLoader the classloader to be used to search the file
 	 * @param resourceName the name of the resource to be pushed
-	 * @throws ApplicationContextBuilderException
+	 * @throws ApplicationContextBuilderException exception occurring while building the application context or underlying class loader
 	 */
 	protected void pushLocalApplicationContext(ClassLoader classLoader, String resourceName) throws ApplicationContextBuilderException {
-		pushLocalApplicationContext(ApplicationContextBuilder.MASTER, classLoader, resourceName);	
+		pushLocalApplicationContext(ApplicationContextBuilder.MASTER, classLoader, resourceName);
 	}
 
 	/**
-	 *Push a new context based on a local file (jar or zip) to the branch specified as argument. See {@link ApplicationContextBuilder} for details
-	 * 
-	 * @param branch the name of the branch on which the context has to be pushed
-	 * @param classLoader the classloader to be used to search the file
+	 * Push a new context based on a local file (jar or zip) to the branch specified as argument. See {@link ApplicationContextBuilder} for details
+	 *
+	 * @param branch       the name of the branch on which the context has to be pushed
+	 * @param classLoader  the classloader to be used to search the file
 	 * @param resourceName the name of the resource to be pushed
-	 * @throws ApplicationContextBuilderException
+	 * @throws ApplicationContextBuilderException exception occurring while building the application context or underlying class loader
 	 */
 	protected void pushLocalApplicationContext(String branch, ClassLoader classLoader, String resourceName) throws ApplicationContextBuilderException {
 		LocalResourceApplicationContextFactory localContext = new LocalResourceApplicationContextFactory(classLoader, resourceName);
-		applicationContextBuilder.pushContext(branch, localContext);			
+		getTokenReservationSession().registerObjectToBeClosedWithSession(applicationContextBuilder.pushContext(branch, localContext));
 	}
 	
 	/**
 	 * Push a new context based on a local folder containing a list of jars to the master branch. See {@link ApplicationContextBuilder} for details
-	 * 
+	 *
 	 * @param libFolder the folder containing the jars to be pushed to the context
-	 * @throws ApplicationContextBuilderException
+	 * @throws ApplicationContextBuilderException exception occurring while building the application context or underlying class loader
 	 */
 	protected void pushLocalFolderApplicationContext(File libFolder) throws ApplicationContextBuilderException {
 		LocalFolderApplicationContextFactory localContext = new LocalFolderApplicationContextFactory(libFolder);
-		applicationContextBuilder.pushContext(ApplicationContextBuilder.MASTER, localContext);			
+		getTokenReservationSession().registerObjectToBeClosedWithSession(applicationContextBuilder.pushContext(ApplicationContextBuilder.MASTER, localContext));

```

Comment:
Duplicated code: "getTokenReservationSession().registerObjectToBeClosedWithSession". One could create method for this